Doc/c-api/time.rst

```diff
@@ -0,0 +1,103 @@
+.. highlight:: c
+
+PyTime C API
+============
+
+.. versionadded:: 3.13
+
+PyTime API
+----------
+
+The PyTime_t API is written to use timestamp and timeout values stored in
+various formats and to read clocks with a resolution of one nanosecond.
+
+The :c:type:`PyTime_t` type is signed to support negative timestamps. The
+supported range is around [-292.3 years; +292.3 years]. Using the Unix epoch
+(January 1st, 1970) as reference, the supported date range is around
+[1677-09-21; 2262-04-11].
+
+
+Types
+-----
+
+.. c:type:: PyTime_t
+
+   Timestamp type with subsecond precision: 64-bit signed integer.
```

Comment:
I think it's worth being specific that `PyTime_t` stores a timestamp in nanoseconds. One of the most common operations will be people taking the difference two `PyTime_t` values, which only makes sense if you known that they represent a time in nanoseconds.

I might say:

A timestamp or duration in nanoseconds represented as a 64-bit signed integer.

The reference point for timestamps depends on the clock used. For example, `PyTime_Time()` returns timestamps relative to the UNIX epoch.